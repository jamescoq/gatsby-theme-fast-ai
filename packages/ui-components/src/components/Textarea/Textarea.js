import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import TextareaControl from '../TextareaControl';
import FormField from '../FormField';
import Text from '../Text';

const Textarea = ({ value, maxCharacters, afterControl, ...otherProps }) => (
	<FormField
		control={TextareaControl}
		kind="textarea"
		value={value}
		{...otherProps}
		afterControl={
			maxCharacters ? (
				<Fragment>
					<Text className="form-group__counter">
						{value.length} / {maxCharacters}
					</Text>
					{afterControl}
				</Fragment>
			) : (
				afterControl
			)
		}
	/>
);

Textarea.propTypes = {
	afterControl: PropTypes.node,
	maxCharacters: PropTypes.number,
	validate: PropTypes.arrayOf(PropTypes.func),
	value: PropTypes.string,
};

Textarea.defaultProps = {
	validate: [],
};

export default Textarea;
