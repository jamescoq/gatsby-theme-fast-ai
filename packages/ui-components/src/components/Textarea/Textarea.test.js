import React from 'react';
import { shallow } from 'enzyme';

import Textarea from './Textarea';

describe('<Textarea />', () => {
	const requiredProps = {};

	it('shallow', () => {
		const wrapper = shallow(<Textarea {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
