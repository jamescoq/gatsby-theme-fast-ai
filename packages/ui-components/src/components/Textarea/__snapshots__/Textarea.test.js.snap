// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> shallow 1`] = `
<withGeneratedId(forwardRef(FormField))
  control={[Function]}
  kind="textarea"
  validate={Array []}
/>
`;
